Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UITransform/UITransformParentingSystem.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using CRDT;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.SDKComponents.SceneUI.Components;
+using ECS.Abstract;
+using ECS.Groups;
+using System.Collections.Generic;
+
+namespace DCL.SDKComponents.SceneUI.Systems.UITransform
+{
+    [UpdateInGroup(typeof(SyncedSimulationSystemGroup))]
+    [UpdateAfter(typeof(UITransformInstantiationSystem))]
+    [LogCategory(ReportCategory.SCENE_UI)]
+    [ThrottlingEnabled]
+    public partial class UITransformParentingSystem : BaseUnityLoopSystem
+    {
+        private readonly Entity sceneRoot;
+        private readonly IReadOnlyDictionary<CRDTEntity, Entity> entitiesMap;
+
+        private UITransformParentingSystem(World world, IReadOnlyDictionary<CRDTEntity, Entity> entitiesMap, Entity sceneRoot) : base(world)
+        {
+            this.sceneRoot = sceneRoot;
+            this.entitiesMap = entitiesMap;
+        }
+
+        protected override void Update(float t)
+        {
+            DoUITransformParentingQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBUiTransform), typeof(UITransformComponent))]
+        private void DoUITransformParenting(in Entity entity, ref PBUiTransform sdkModel, ref UITransformComponent uiTransformComponent)
```

Comment:
There is logic missing that removes a child from the old parent. Currently, you only `Add` to `Children` and never remove

Reply:
Good point! You're right, I didn't take into account the child removal cases.

Done: https://github.com/decentraland/unity-explorer/pull/204/commits/8a1cc98ec2c8cd9ec259b9abf4caa5eac5f9ca2a